pubtools/pulplib/_impl/fake/controller.py

```diff
@@ -79,3 +79,18 @@ def publish_history(self):
                 of this publish
         """
         return self.client._publish_history[:]
+
+    @property
+    def upload_history(self):
+        """A list of upload tasks triggered via this client.
+
+        Each element of this list is a named tuple with the following attributes,
+        in order:
+
+            ``repository``:
```

Comment:
Documentation doesn't match reality.  The namedtuple you declared actually has a "repo_id" and not a "repository".

In this case, I think the documentation is better and the code should be fixed.  Publish returns a repository object and not a repo_id, why should Upload be inconsistent with that?  Can you please fix that to have a repository object as well?

Reply:
copy-pasted and wrong :(